src/status_im/contexts/wallet/send/input_amount/view.cljs

```diff
@@ -199,22 +230,40 @@
          [quo/token-input
           {:container-style style/input-container
            :token           token-symbol
-           :currency        current-currency
-           :crypto-decimals crypto-decimals
+           :currency        fiat-currency
+           :crypto-decimals (min token-decimals 6)
            :error?          (controlled-input/input-error input-state)
            :networks        (seq token-networks)
-           :title           (i18n/label :t/send-limit
-                                        {:limit (make-limit-label current-limit current-currency)})
+           :title           (i18n/label
+                             :t/send-limit
+                             {:limit (if @crypto-currency?
+                                       (make-limit-label-crypto current-limit token-symbol)
+                                       (make-limit-label-fiat current-limit
+                                                              (get constants/currency-symbols
+                                                                   fiat-currency)))})
            :conversion      conversion-rate
            :show-keyboard?  false
-           :value           (controlled-input/input-value input-state)
-           :on-swap         #(reset! crypto-currency? %)
+           :value           input-amount
+           :on-swap         (fn [swap-to-crypto-currency?]
+                              (set-just-toggled-mode? true)
+                              (reset! crypto-currency? swap-to-crypto-currency?)
+                              (set-input-state
+                               #(controlled-input/set-input-value
+                                 %
+                                 (let [value     (controlled-input/input-value %)
```

Comment:
Yes, better to give a name to that var.

@briansztamfater I prefer to follow the approach @ilmotta  once shared with me: use the shorthand (`#()`) only when its a simple one liner function, otherwise prefer to use `(fn [])` because it's clearer what is received or even if it's empty it's clearer than looking for a `%` in the function.